rustls-symcrypt/src/signer.rs

```diff
@@ -1,22 +1,394 @@
+        // instead of the message to digest. This function does not take a precomputed digest; instead, sign calculates the digest itself.
+
+        match self.scheme {
+            SignatureScheme::RSA_PKCS1_SHA256 => {
+                let hashed_message = sha256(message);
+                match self.key.pkcs1_sign(&hashed_message, HashAlgorithm::Sha256) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PKCS1_SHA384 => {
+                let hashed_message = sha384(message);
+                match self.key.pkcs1_sign(&hashed_message, HashAlgorithm::Sha384) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PKCS1_SHA512 => {
+                let hashed_message = sha512(message);
+                match self.key.pkcs1_sign(&hashed_message, HashAlgorithm::Sha512) {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PSS_SHA256 => {
+                let hashed_message = sha256(message);
+                match self
+                    .key
+                    .pss_sign(&hashed_message, HashAlgorithm::Sha256, SHA256_RESULT_SIZE)
+                {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PSS_SHA384 => {
+                let hashed_message = sha384(message);
+                match self
+                    .key
+                    .pss_sign(&hashed_message, HashAlgorithm::Sha384, SHA384_RESULT_SIZE)
+                {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            SignatureScheme::RSA_PSS_SHA512 => {
+                let hashed_message = sha512(message);
+                match self
+                    .key
+                    .pss_sign(&hashed_message, HashAlgorithm::Sha512, SHA512_RESULT_SIZE)
+                {
+                    Ok(signature) => Ok(signature),
+                    Err(e) => Err(Error::General(format!("failed to sign message: {}", e))),
+                }
+            }
+            _ => {
+                return Err(Error::General(
+                    "unsupported RSA-PKCS1 signature scheme".into(),
+                ));
+            }
+        }
+    }
+
+    fn scheme(&self) -> SignatureScheme {
+        self.scheme
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
Why is #[allow(dead_code)] here? does this fail cargo doc tests?

Reply:
I had some unused parameter in previous code, this is to get it compile I think. It is removed now.